articles/fleetctl.md

```diff
@@ -20,6 +20,16 @@ sudo npm install -g fleetctl
 
 To install fleetctl on Windows or Linux, download the fleectl binary here on [GitHub](https://github.com/fleetdm/fleet/releases).
 
+### Additional dependencies
+
+To use `fleetctl package` for platforms other than Windows, you'll need to run `fleetctl` from macOS or Linux.
+
+To use `fleetctl package` to create macOS installers (.pkg files), if you're using Linux you'll need Docker, as we use utilities in the `fleetdm/bomutils` Docker image there.
+
+To use `fleetctl package` to create Windows installers (.msi files), you'll need Docker if you aren't on an Apple Silicon Mac. This is because we use the `fleetdm/wix` Docker image to provide the [WiX](https://github.com/wixtoolset/wix3) dependency on most platforms. For Apple Silicon Macs, we automatically install WiX locally when needed.
```

Comment:
Hey @iansltx, great docs.

What do you think about moving these to the Enroll hosts guide instead? I'm thinking at the bottom of the top "CLI" section. Above "Example":

![Screenshot 2024-10-15 at 10 40 15 AM](https://github.com/user-attachments/assets/58056b4f-0d13-4eaf-9617-b554068f4e55)

As a user I would want to know about these dependencies when I'm trying to generate a fleetd to enroll hosts. I think the "fleetctl" guide is more of a general, getting started intro to the fleectl CLI tool.

Reply:
Maybe we move the `fleetctl package` requirements to the article above, while linking to them from the generic docs, and keep `fleetctl preview` as is?

Reply:
> linking to them from the generic docs

You mean, once we move, link to the `fleetctl package` dependencies in the "Enroll hosts" guide from the "fleetctl" guide? I'm not sure we need to. For sake of less.

We don't mention `fleetctl package` in the "fleetctl guide".

That said, now I'm looking at the entire "Add hosts" flow. I think we should link the user to "Enroll hosts" instead of "fleetctl" guide in the "Add hosts" modal here:

![Screenshot 2024-10-15 at 10 56 03 AM](https://github.com/user-attachments/assets/d2d60ed3-5274-48c6-954a-69cd38260360)

That way, the user sees instructions for `fleetctl package` and it's dependencies. If they don't have `fleetctl` there's already a link in the "Enroll hosts" guide here:

![Screenshot 2024-10-15 at 10 56 54 AM](https://github.com/user-attachments/assets/6374fa76-3a73-4f20-a3c0-41cc9f84be17)

What do you think?

Reply:
So, linking to that doc just spits out information that's already baked into the command being shown in the UI, so basically the only reason someone is going to click that link is "I don't have fleetctl yet," which means that if they land on the enrollment page they're going to have to immediately click through to the installation instructions anyway, then go _back_ to the enrollment page to see what other dependencies they need.

If we flip that flow so that the link in "Add hosts" lands right at https://fleetdm.com/guides/fleetctl#installing-fleetctl, then we link from _there_ to the packaging instructions that would have the additional system requirements, users wouldn't have to backtrack to get everything they need, and would end up on the host enrollment docs, which would include more info on customizing the CLI snippet they're seeing in the UI.

Let me know if I'm getting the expected user flow wrong here, but my guess is that enrolling hosts is so early in the onboarding experience that if someone's clicking that link it's because they're five minutes into setting up Fleet and haven't installed fleetctl yet, and will need to create packaging after that.

I'll also go out on a limb and say that the majority of visits to that fleetctl guide are from the learn-more-about link that's only exposed in the UI above (if we have page view and/or referrer metrics we should be able to confirm this).

Reply:
> it's because they're five minutes into setting up Fleet and haven't installed fleetctl yet, and will need to create packaging after that.

> the majority of visits to that fleetctl guide are from the learn-more-about link that's only exposed in the UI above

I think you're right!

> If we flip that flow so that the link in "Add hosts" lands right at https://fleetdm.com/guides/fleetctl#installing-fleetctl, then we link from there to the packaging instructions that would have the additional system requirements, users wouldn't have to backtrack to get everything they need, and would end up on the host enrollment docs, which would include more info on customizing the CLI snippet they're seeing in the UI.

I think you're on the right track here! This journey makes sense to me.

Reply:
Cool. Was going to hop on design review tmw to hash this out, but I think I have what I need to do another rev of this async, and if this isn't ✅ by design review I'll swing by to hammer out any final tweaks.

One nuance here is that the existing learn-more-about on fleetctl doesn't land on the install instructions, but given that this is a solve for users new to Fleet I think it's reasonable to add a new redirect link that goes to the correct anchor and leave "fleetctl" as-is so that, if we wind up using that link elsewhere (we aren't right now) it doesn't go to what's probably the wrong place (our other refs to fleetctl docs in the app aren't about installing it).

Reply:
> I think it's reasonable to add a new redirect link that goes to the correct anchor and leave "fleetctl" as-is so that, if we wind up using that link elsewhere (we aren't right now) it doesn't go to what's probably the wrong place (our other refs to fleetctl docs in the app aren't about installing it).

@iansltx agreed 💯